code/modules/mob/living/carbon/carbon_organs.dm

```diff
@@ -61,3 +59,19 @@
 	//Check if we should die
 	if(species.is_vital_organ(src, O))
 		death()
+
+//Called by surgeries when detaching an organ during organ detach surgery
+/mob/living/carbon/surgical_detach_organ(var/obj/item/organ/O, var/obj/item/organ/external/parent)
```

Comment:
This is _exactly_ why I was against moving the removal and addition procs onto the mob.

Reply:
Its more of an helper than anything. Even before the organ changes, it was just a long snippet copy pasted a million times to do that.
Also, I'm not entirely sure what you're really getting to anyways?

Reply:
Metastatizing new specialized procs on the mob is exactly what I was expecting to happen from your changes to scope and encapsulation of organ behavior. It's going to be more brittle, it's going to harder to override for specific organs, everything I spoke about at length prior to your human init PR being merged.

It's moot, but I felt obligated to say something. You can disregard this comment.

Reply:
I'm gonna be honest here. Its hard to be more brittle than it was already. And I actually noticed it was holding up pretty well in that regard, even seeing an improvement.

Also, I did try to integrate more of your approach from your PR, but a lot of it didn't work, or didn't take things into account. It didn't seem like you tested/investigated a lot of it before. (EX: things like mobs not having eyes for example because get_organ would be used in too many cases and override organ zones and return the head when passed BP_EYES) 
It also touched to so many different systems that I didn't feel like spending time to fill in the blanks, and deal with the massive amount of testing myself, since that's was way out of scope for me. And you should keep in mind my changes were never meant to be the be all, end all of organs. I fully expected there would be a need for further refactors as humans are overhauled. If you think you can do it better, by all means be my guest! I'm just doing what I can with what I got.

Also, concerning your comment, I integrated the surgery behavior into add_organ and remove_organ now via a boolean param. I think its less clear and less overridable than having a proc to handle it, but I guess it means less specialized procs. 
Will be pushing once I test everything.